### PR TITLE
Factor out execution

### PR DESCRIPTION
**Pull request by janfreyberg**
This moves the execution of notebooks to a separate file.

### CODE REVIEW

File: faculty_mill/faculty_reporter.py [hunk 1]
```diff
@@ -9,9 +8,9 @@
 import click
 import sherlockml
 import sml.auth
-from papermill.cli import papermill
 
 from .version import print_version
+from .execute import run
```
Comment (liamcoatman) [inline faculty_mill/faculty_reporter.py]: I think the concensus is that `faculty_mill.execute` is better than using the relative import (although I'm not sure why).

File: faculty_mill/faculty_reporter.py [hunk 2]
```diff
@@ -77,23 +76,7 @@ def main(
...
-                parent=click_context,
-            )
-
-            papermill.invoke(papermill_click_context)
-        else:
-            shutil.copy(input_path, output_path)
+        output_path = run(notebook, directory, execute, click_context)
 
         reports = {
             report.name: report for report in report_client.list(PROJECT_ID)
```
Comment (liamcoatman) [inline faculty_mill/faculty_reporter.py]: I don't think PROJECT_ID (and USER_ID) should be capitalized

File: faculty_mill/faculty_reporter.py (hunk 1 above)
Comment (janfreyberg) [reply faculty_mill/faculty_reporter.py]: I would be fairly keen to keep the relative import because it's nicer

Comment (liamcoatman) [reply faculty_mill/faculty_reporter.py]: Fine. In that case I think there was somewhere else that used the `faculty_mill.` style you might want to change for consistency

Comment (janfreyberg) [reply faculty_mill/faculty_reporter.py]: yep, will do!

File: faculty_mill/faculty_reporter.py (hunk 2 above)
Comment (janfreyberg) [reply faculty_mill/faculty_reporter.py]: I believe this is addressed in the other PR

**Pull request merged**